src/nupic/algorithms/TemporalMemory.cpp

```diff
@@ -671,11 +671,29 @@ UInt TemporalMemory::columnForCell(const CellIdx cell) const {
   return cell / cellsPerColumn_;
 }
 
+
+SDR TemporalMemory::cellsToColumns(const SDR& cells) const {
+  NTA_CHECK(cells.size == numberOfCells()) 
+	  << "cells.size " << cells.size << " must match TM::numberOfCells() " << numberOfCells();
+
+  SDR cols({numColumns_});
+  auto& dense = cols.getDense();
+  for(const auto cell : cells.getSparse()) {
+    const auto col = columnForCell(cell);
+    dense[col] = 1;
+  }
+  cols.setDense(dense);
+
+  NTA_ASSERT(cols.size == numColumns_); 
+  return cols;
+}
+
+
 vector<CellIdx> TemporalMemory::cellsForColumn(CellIdx column) { 
   const CellIdx start = cellsPerColumn_ * column;
   const CellIdx end = start + cellsPerColumn_;
 
-  vector<CellIdx> cellsInColumn;
+  vector<CellIdx> cellsInColumn(cellsPerColumn_);
```

Comment:
Did you mean to say:
`vector<CellIdx> cellsInColumn; cellsInColumn.reserve(cellsPerColumn_);`

I think that as this is, it will initialize the list with cellsPerColumn_ many zeros, which are not ever removed, and are not part of the intended result.